climada/hazard/tc_tracks.py

```diff
@@ -1587,17 +1648,88 @@ def ibtracs_track_agency(ds_sel):
     track_agency_ix : xarray.DataArray of ints
         For each entry in `ds_sel`, the agency to use, given as an index into `agency_pref`.
     """
-    agency_pref = IBTRACS_AGENCIES.copy()
+    agency_pref = ["wmo"] + IBTRACS_AGENCIES
     agency_map = {a.encode('utf-8'): i for i, a in enumerate(agency_pref)}
     agency_map.update({
         a.encode('utf-8'): agency_map[b'usa'] for a in IBTRACS_USA_AGENCIES
     })
     agency_map[b''] = agency_map[b'wmo']
     agency_fun = lambda x: agency_map[x]
-    track_agency = ds_sel.wmo_agency.where(ds_sel.wmo_agency != '', ds_sel.usa_agency)
-    track_agency_ix = xr.apply_ufunc(agency_fun, track_agency, vectorize=True)
+    if "track_agency" not in ds_sel.data_vars.keys():
+        ds_sel['track_agency'] = ds_sel.wmo_agency.where(ds_sel.wmo_agency != b'',
+                                                         ds_sel.usa_agency)
+    track_agency_ix = xr.apply_ufunc(agency_fun, ds_sel.track_agency, vectorize=True)
     return agency_pref, track_agency_ix
 
+def ibtracs_add_official_variable(ibtracs_ds, var, add_3h=False):
+    """Add variables for officially reported values to IBTrACS dataset
+
+    This function adds new variables to the xarray `ibtracs_ds` that contain values of the
+    specified TC variable `var` that have been reported by the officially responsible agencies.
+    For example, if `var` is "wind", there will be a new variable "official_wind" and, if `add_3h`
+    is True, a variable "official_3h_wind".
+
+    Parameters
+    ----------
+    ibtracs_ds : xarray.Dataset
+        Subselection of original IBTrACS NetCDF dataset.
+    var : str
+        Name of variable for which to add an "official" version.
+    add_3h : bool, optional
+        Optionally, add an "official_3h" version where also 3-hourly data by the officially
```

Comment:
I updated the docstrings to be more verbose about this, hopefully that helps. Note that there are "officially responsible agencies" and there are "official reporting times" or "official time steps". Those are two different things. An agency that is officially responsible at some point in time for a specific storm will always report at the official time steps. But it might as well report at additional time steps.